tests/io_components/test_dataset.py

```diff
@@ -120,10 +116,7 @@ def test_dataset_get_indices_as_dataframe_no_index(dataset):
 def test_dataset_get_indices_as_dataframe_with_index(dataset_with_index, store_session):
     assert not dataset_with_index.primary_indices_loaded
     df = dataset_with_index.get_indices_as_dataframe()
-    if Version(pd.__version__) < Version("2.0.0.dev0"):
-        empty_index = pd.Index([], name="partition")
-    else:
-        empty_index = pd.Index([], name="partition", dtype="int64")
+    empty_index = pd.Index([], name="partition")
```

Comment:
I think this check may no longer be required. It seems that the behaviour of `pd.Index` in pandas 2.1.0, 2.1.1 and 2.1.2 is the same as that in 1.5.3, whereas it was different in pandas 2.0.0-2.0.3. Making these two tests more similar to how they looked before #69 appears to get them passing again.

I haven't yet found the commits which originally caused this change in behaviour in pandas 2.0 and then reverted it in pandas 2.1, but if that is of interest, I could probably do so with git bisect.